Framework/DataHandling/src/PulseIndexer.cpp

```diff
@@ -0,0 +1,86 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "MantidDataHandling/PulseIndexer.h"
+#include <sstream>
+#include <utility>
+
+namespace Mantid::DataHandling {
+PulseIndexer::PulseIndexer(std::shared_ptr<std::vector<uint64_t>> event_index, const std::size_t firstEventIndex,
+                           const std::size_t numEvents, const std::string &entry_name)
+    : m_event_index(std::move(event_index)), m_firstEventIndex(firstEventIndex), m_numEvents(numEvents),
+      m_entry_name(entry_name) {
+  m_numPulses = m_event_index->size();
+}
+
+/**
+ * This performs a linear search because it is much more likely that the index
+ * to look for is at the beginning.
+ */
+size_t PulseIndexer::getFirstPulseIndex(const size_t event_index) const {
```

Comment:
I don't understand why this function's name includes the word "Pulse" because I read nothing in the body of the function that refers to pulse indexes. I only read "event indexes".